astropy/units/format/utils.py

```diff
@@ -41,22 +55,22 @@ def get_grouped_by_powers(bases, powers):
     return positive, negative
 
 
-def split_mantissa_exponent(v, format_spec=".8g"):
+def split_mantissa_exponent(v: Real, format_spec: str = ".8g") -> tuple[str, str]:
     """
     Given a number, split it into its mantissa and base 10 exponent
     parts, each as strings.  If the exponent is too small, it may be
     returned as the empty string.
 
     Parameters
     ----------
-    v : float
+    v : `numbers.Real`
```

Comment:
For other docstrings, we just use "number", which numpy-doc links properly; e.g.,
https://docs.astropy.org/en/latest/api/astropy.units.quantity.allclose.html#astropy.units.quantity.allclose